src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2436,7 +2447,168 @@ public void UpdateIsNullableEnabledForConstructorsAndFields(bool useStatic, bool
+                Debug.Assert(!staticInitializers.IsDefault);
+                Debug.Assert(!instanceInitializers.IsDefault);
+                Debug.Assert(!indexerDeclarations.IsDefault);
+
+                Debug.Assert(!nonTypeNonIndexerMembers.Any(s => s is TypeSymbol));
+                Debug.Assert(!nonTypeNonIndexerMembers.Any(s => s.IsIndexer()));
+                Debug.Assert(!nonTypeNonIndexerMembers.Any(s => s.IsAccessor() && ((MethodSymbol)s).AssociatedSymbol.IsIndexer()));
+
+                this.NonTypeNonIndexerMembers = nonTypeNonIndexerMembers;
+                this.StaticInitializers = staticInitializers;
+                this.InstanceInitializers = instanceInitializers;
+                this.IndexerDeclarations = indexerDeclarations;
+                this.StaticInitializersSyntaxLength = staticInitializersSyntaxLength;
+                this.RecordDeclarationWithParameters = recordDeclarationWithParameters;
+                this.InstanceInitializersIndexForRecordDeclarationWithParameters = instanceInitializersIndexForRecordDeclarationWithParameters;
+                this.IsNullableEnabledForInstanceConstructorsAndFields = isNullableEnabledForInstanceConstructorsAndFields;
+                this.IsNullableEnabledForStaticConstructorsAndFields = isNullableEnabledForStaticConstructorsAndFields;
+            }
+        }
+
+        private sealed class MembersAndInitializersBuilder
+        {
+            public ArrayBuilder<Symbol> NonTypeNonIndexerMembers { get; set; }
+            public ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> InstanceInitializers;
+            private bool IsNullableEnabledForInstanceConstructorsAndFields;
+            private bool IsNullableEnabledForStaticConstructorsAndFields;
+
+            public MembersAndInitializersBuilder(DeclaredMembersAndInitializers declaredMembersAndInitializers)
+                : this(
+                    declaredMembersAndInitializers.NonTypeNonIndexerMembers,
+                    declaredMembersAndInitializers.InstanceInitializers,
+                    declaredMembersAndInitializers.IsNullableEnabledForInstanceConstructorsAndFields,
+                    declaredMembersAndInitializers.IsNullableEnabledForStaticConstructorsAndFields)
+            {
+            }
+
+            private MembersAndInitializersBuilder(
+                ImmutableArray<Symbol> nonTypeNonIndexerMembers,
+                ImmutableArray<ImmutableArray<FieldOrPropertyInitializer.Builder>> instanceInitializers,
+                bool isNullableEnabledForInstanceConstructorsAndFields,
+                bool isNullableEnabledForStaticConstructorsAndFields)
+            {
+                Debug.Assert(!nonTypeNonIndexerMembers.IsDefault);
+                Debug.Assert(!instanceInitializers.IsDefault);
+                Debug.Assert(instanceInitializers.All(g => !g.IsDefault));
+
+                this.NonTypeNonIndexerMembers = getAndFillBuilder(nonTypeNonIndexerMembers);
```

Comment:
>getAndFillBuilder(nonTypeNonIndexerMembers); [](start = 48, length = 44)

This doesn't feel right. I don't think we should copy content into this builder. We should simply combine members later, if necessary. Otherwise we should reuse existing immutable array.

Reply:
The synthesized constructors are added at the end, but the record synthesized members are added at the beginning...
Making a new builder is simple and allows to maintain that existing order with minimal gymnastics.

---
In reply to: [562783607](https://github.com/dotnet/roslyn/pull/50650#discussion_r562783607) [](ancestors = 562783607)

Reply:
> The synthesized constructors are added at the end, but the record synthesized members are added at the beginning...
Making a new builder is simple and allows to maintain that existing order with minimal gymnastics.

Both are not very common, especially record initializers because records are not common yet. I believe in most common case we are not going to add anything. I think we can easily change what we do for the synthesized constructor. Even if we don't want to do that, the logic doesn't feel prohibitively complex. 

---
In reply to: [562828146](https://github.com/dotnet/roslyn/pull/50650#discussion_r562828146) [](ancestors = 562828146,562783607)